UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -826,6 +835,25 @@ public virtual void AgentAction(float[] vectorAction, string textAction)
         {
         }
 
+        /// <summary>
+        /// Specifies the agent behavior at every step based on the provided
+        /// action.
+        /// </summary>
+        /// <param name="vectorAction">
+        /// Vector action. Note that for discrete actions, the provided array
+        /// will be of length 1.
+        /// </param>
+        /// <param name="textAction">Text action.</param>
+        /// <param name="customAction">
+        /// A custom action, defined by the user as custom protobuffer message. Useful if the action is hard to encode
+        /// as either a flat vector or a single string.
+        /// </param>
+        public virtual void AgentAction(float[] vectorAction, string textAction, CommunicatorObjects.CustomAction customAction)
```

Comment:
This will change our demo environments code. We will need to update them before merging

Reply:
Actually no because of L854